stdlib/public/core/ExistentialCollection.swift.gyb

```diff
@@ -718,27 +718,20 @@ public func < (lhs: AnyIndex, rhs: AnyIndex) -> Bool {
 /// A protocol for `AnyCollection<Element>`,
 /// `AnyBidirectionalCollection<Element>`, and
 /// `AnyRandomAccessCollection<Element>`.
-///
-/// This protocol can be considered an implementation detail of the
-/// `===` and `!==` implementations for these types.
 public protocol AnyCollectionProtocol : Collection {
   /// Identifies the underlying collection stored by `self`. Instances
   /// copied from one another have the same `_underlyingCollectionID`.
   var _underlyingCollectionID: ObjectIdentifier { get }
-}
 
-/// Returns `true` iff `lhs` and `rhs` store the same underlying collection.
-public func === <
-  L : AnyCollectionProtocol, R : AnyCollectionProtocol
->(lhs: L, rhs: R) -> Bool {
-  return lhs._underlyingCollectionID == rhs._underlyingCollectionID
+  /// Returns `true` iff `self` and `other` use the same `_box` to store their
+  /// underlying collections.
+  func _isKnownToBeEqual<C: AnyCollectionProtocol>(to other: C) -> Bool
 }
 
-/// Returns `false` iff `lhs` and `rhs` store the same underlying collection.
-public func !== <
-  L : AnyCollectionProtocol, R : AnyCollectionProtocol
->(lhs: L, rhs: R) -> Bool {
-  return lhs._underlyingCollectionID != rhs._underlyingCollectionID
+extension AnyCollectionProtocol {
+  public func _isKnownToBeEqual<C: AnyCollectionProtocol>(to other: C) -> Bool {
```

Comment:
Missed these, sorry! 

@gribozavr If we move `AnyCollectionProtocol` out of the library, we wouldn't be able to test this case without making `_box` public:

```swift
let a = AnyCollection([1, 2, 3, 4, 5])
let b = AnyRandomAccessCollection(a)!
// a. _storesSameUnderlyingCollection(b) == true
```
Proposal: add an underscore to `AnyCollectionProtocol` and change `_underlyingCollectionID` to `_boxID`. Thumbs up/down?